odoo/model.go

```diff
@@ -0,0 +1,92 @@
+package odoo
+
+import "fmt"
+
+// SearchReadModel is used as "params" in requests to "dataset/search_read" endpoints.
+type SearchReadModel struct {
+	Model  string   `json:"model,omitempty"`
+	Domain []Filter `json:"domain,omitempty"`
+	Fields []string `json:"fields,omitempty"`
+	Limit  int      `json:"limit,omitempty"`
+	Offset int      `json:"offset,omitempty"`
+}
+
+// Filter to use in queries, usually in the format of
+// [predicate, operator, value], eg ["employee_id.user_id.id", "=", 123]
+type Filter interface{}
+
+// Method identifies the type of write operation.
+type Method string
+
+const (
+	// MethodWrite is used to update existing records.
+	MethodWrite Method = "write"
+	// MethodCreate is used to create new records.
+	MethodCreate Method = "create"
+	// MethodRead is used to read records.
+	MethodRead Method = "read"
```

Comment:
Currently, it's unused. However there actually is a read endpoint with method "read" (alternative to `search_read`), in case someone uses this (https://www.odoo.com/documentation/8.0/api_integration.html)

however, I'm fine with removing unused code

Reply:
Aah. I thought that was another method for the `WriteModel`. Which would have been weird. That makes sense and we can leave it in IMO.